### PR TITLE
16.0 sp global filters pro rar

### PR DESCRIPTION
**Pull request by rrahir**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/spreadsheet/static/tests/global_filters/global_filters_model_test.js [hunk 1]
```diff
@@ -1627,52 +1620,99 @@ QUnit.module("spreadsheet > Global filters model", {}, () => {
...
+    });
+
+    QUnit.test("field matching is removed when an Odoo chart is deleted", async function (assert) {
+        const { model } = await createSpreadsheetWithChart({ type: "odoo_pie" });
+        const sheetId = model.getters.getActiveSheetId();
+        const [chartId] = model.getters.getChartIds(sheetId);
+        await addGlobalFilter(model, LAST_YEAR_FILTER, {
+            chart: { [chartId]: { chain: "date", type: "date" } },
+        });
+        // await addGlobalFilter(
```
Comment (LucasLefevre) [inline addons/spreadsheet/static/tests/global_filters/global_filters_model_test.js:1694]: coucou comment

File: addons/spreadsheet/tests/validate_spreadsheet_data.py [hunk 2]
```diff
@@ -120,17 +120,36 @@ def filter_fields(data):
...
-    for filter_definition in data["globalFilters"]:
-        for pivot_id, matching in filter_definition.get("pivotFields", dict()).items():
-            model = data["pivots"][pivot_id]["model"]
-            fields_by_model[model].add(matching["field"])
-        for list_id, matching in filter_definition.get("listFields", dict()).items():
-            model = data["lists"][list_id]["model"]
-            fields_by_model[model].add(matching["field"])
-        for chart_id, matching in filter_definition.get("graphFields", dict()).items():
-            chart = next((chart for chart in charts if chart["id"] == chart_id), None)
+    odooVersion = data.get("odooVersion", 1)
```
Comment (LucasLefevre) [inline addons/spreadsheet/tests/validate_spreadsheet_data.py:123]:
```suggestion
    odoo_version = data.get("odooVersion", 1)
```

**Review by pro-odoo — APPROVED** [review]
robodoo rebase-ff r+

**Comment by rrahir** [comment]
@robodoo r+

**Pull request closed without merging**